experimental/offline_attestation/shared/src/lib.rs

```diff
@@ -0,0 +1,152 @@
+//
+// Copyright 2022 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+//! Shared components used by both the client and server for offline attestation.
+
+use anyhow::anyhow;
+use serde::{Deserialize, Serialize};
+pub use tink_core::keyset::Handle;
+pub use tink_hybrid::init;
+
+const ENCRYPTION_CONTEXT: &[u8] = b"Offline Attestation v0.1";
+
+/// An encrypted request sent from the client to the server.
+#[derive(Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct EncryptedRequest {
+    /// The encrypted payload.
+    ///
+    /// Contains the raw output from Tink hybrid encryption.
+    pub ciphertext: Vec<u8>,
+
+    /// The public key that should be used for encrypting the response.
+    ///
+    /// Contains the binary-serialised form of the Tink keyset representing the public key.
+    pub response_public_key: Vec<u8>,
```

Comment:
I'm curious why you went with this approach (which leaves the public key unencrypted) as opposed to encrypting the public key as part of the payload directly. What are the trade offs between the two options?

Reply:
The idea is that the client would generate a new key pair for every request/response, so the public key does not contain any usable information and should not enable tracking. Encrypting it does not seem to be required.

I considered a version where a symmetric key was included in the encrypted payload, but Tink tries really hard to stop us from getting to the underlying symmetric keys. If we do a future implemenation using lower-level crypto primitives we could consider that option again.

Reply:
Makes sense, thanks.